docs/VISION.md

```diff
@@ -0,0 +1,137 @@
+
+0. Open
+1. Correct
+2. Fast
+3. HTTP/\*
+4. Flexible
+5. Understandable
+
+There's a lot more detail about each in [TENETS](./TENETS.md).
+
+## Use Cases
+
+Who are the *users* of hyper? How would they use hyper?
+
+### Low-Level Client Library (curl, reqwest, aws-sdk)
+
+These client libraries care that hyper is **Flexible**, since they are expressing their own opinion on how a more-featured HTTP client should act. This includes opinions on connection establishment, management, pooling, HTTP version options, and even runtimes.
+
+curl's main reason for using hyper is that it is **Safe**.
+
+### Web Server Frameworks (deno, axum)
+
+These are using hyper's server feature to expose a different, higher-level API to users. Besides the obvious requirements, these require that hyper is **Fast**. Servers are costly, handling more requests faster is important to them.
+
+That hyper is **Flexible** is also important, in that it needs to be flexible enough for them to build a server framework, and allow them to express their own opinions about API to their users.
+
+### Services and Proxies (linkerd, cloudflare, fastly)
+
+These are using hyper directly, likely both the client and server, in order to build efficient and powerful services, applications, and tools for their end users. They care greatly that hyper is **Correct**, since web traffic can stretch the limits of what is valid HTTP, and exercise less-common parts of the specifications.
+
+They also require hyper to be **Fast**, for similar reasons that the web server frameworks do.
+
+### New Rust Web Developers
+
+These are developers who are either new to Rust, or new to web servers, and have reached for hyper to start with.
+
+It's likely that these users don't have strong opinions about how an HTTP server or client should work, just that it _should_ handle all the things they normally assume it would. For these users, it would be best to quickly help them compare their own expectations with hyper's capabitilities, and may suggest reaching for higher-level, _easier_ libraries instead.
+
+Those that stick around after that recommendation are users that wish both to learn at a lower level, and to pick and choose what batteries they plug in to hyper as they move along. While they do care about the other tenets, that hyper is **Understandable** is of extra importance to them.
+
+## The Library
+
+So with all that context in mind, what does hyper, the library, actually look like? This doesn't highlight what _is_ and _isn't_ present. What currently needs to change to reach this vision is left to individual version roadmaps.
+
+### Layers
+
+In all cases, a user brings their own runtime and IO to work with hyper. The IO is provided to hyper, and hyper acts on top of it. hyper returns `Future`s that the user then decides how to poll, likely involving their runtime options.
```

Comment:
Could you add something here, and possibly in the tenets, about the degree to which hyper provides optional runtime integration? For instance, hyper currently has an optional `runtime` feature; what criteria would hyper use to evaluate proposals for additional such features? (e.g. maintainability, not compromising higher-priority tenets) Similarly, what criteria would be used to evaluate (optional) integration with various different versions of traits such as `AsyncRead`/`AsyncWrite`?

These seem like questions people will reasonably ask, for which it'd be helpful to have a reference in the tenets and vision.